src/Evolution/Systems/GeneralizedHarmonic/UpwindPenaltyCorrection.hpp

```diff
@@ -0,0 +1,236 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/FaceNormal.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Protocols.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace GeneralizedHarmonic {
+/*!
+ * \brief Computes the generalized-harmonic upwind multipenalty boundary
+ * correction.
+ *
+ * This implements the upwind multipenalty boundary correction term
+ * \f$D_\alpha\f$. The general form is given by:
+ *
+ * \f{align*}{
+ *   \label{eq:pnpm upwind boundary term characteristics}
+ *   D_\beta =
+ *   T_{\beta\hat{\beta}}^{\mathrm{ext}}
+ *   \Lambda^{\mathrm{ext},-}_{\hat{\beta}\hat{\alpha}}
+ *   v^{\mathrm{ext}}_{\hat{\alpha}}
+ *   -T_{\beta\hat{\beta}}^{\mathrm{int}}
+ *   \Lambda^{\mathrm{int},-}_{\hat{\beta}\hat{\alpha}}
+ *   v^{\mathrm{int}}_{\hat{\alpha}},
+ * \f}
+ *
```

Comment:
I suggest clarifying that hatted indexes index the characteristic variables, while unhatted indexes index the evolved variables.